File: 71-simplify-path/simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        s = path.split("/")
        stack = []

        for c in s:
            if c == "..":
                if stack:
                    stack.pop()
            elif c in (".", ""):
                continue
            else:
                stack.append(c)
        return "/" + "/".join(stack)

File: 71-simplify-path/simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        stack<string> st;
        for (int i = 0; i < path.size(); ++i) {
            if (path[i] == '/') {
                continue;
            }

            string tmp;
            while (i < path.size() && path[i] != '/') {
                tmp += path[i];
                ++i;
            }

            if (tmp == "..") {
                if (!st.empty()) {
                    st.pop();
                }
            } else if (tmp == ".") {
                continue;
            } else {
                st.push(tmp);
            }
        }

        if (st.empty()) {
            return "/";
        }

        string res;
        while (!st.empty()) {
            res = '/' + st.top() + res;
            st.pop();
        }

        return res;
    }
};